### PR TITLE
testing podman

### PR DESCRIPTION
**Pull request by tsorya**

### CODE REVIEW

File: skipper/utils.py [hunk 1]
```diff
@@ -137,3 +137,20 @@ def image_to_dockerfile(image):
 
 def dockerfile_to_image(dockerfile):
     return dockerfile.replace('Dockerfile.', '')
+
+
+def get_runtime_command():
+    global CONTAINER_RUNTIME_COMMAND
+    if not CONTAINER_RUNTIME_COMMAND:
+        print("AAAAAAAAAAAAAAAAAA")
```
Comment (romfreiman) [inline skipper/utils.py:145]: AAAAA

File: skipper/utils.py [hunk 2]
```diff
@@ -137,3 +137,20 @@ def image_to_dockerfile(image):
 
 def dockerfile_to_image(dockerfile):
     return dockerfile.replace('Dockerfile.', '')
+
+
+def get_runtime_command():
+    global CONTAINER_RUNTIME_COMMAND
```
Comment (romfreiman) [inline skipper/utils.py:143]: better take it from ENV or config? Why hardcoded in the code?
Also, you should assume the server have both docker and podman installed

File: skipper/utils.py (hunk 1 above)
Comment (tsorya) [reply skipper/utils.py:145]: test :)

File: skipper/utils.py (hunk 2 above)
Comment (tsorya) [reply skipper/utils.py:143]: are u sure u cand have 2 of them?

**Comment by romfreiman** [comment]
yes

File: skipper/utils.py (hunk 1 above)
Comment (eranco74) [inline skipper/utils.py:145]: AAAA?

File: skipper/utils.py [hunk 3]
```diff
@@ -137,3 +137,20 @@ def image_to_dockerfile(image):
...
+    if not CONTAINER_RUNTIME_COMMAND:
+        print("AAAAAAAAAAAAAAAAAA")
+        cmd = "if  [ -x \"$(command -v docker)\" ];then echo docker ; else echo podman;fi"
+        runtime = str(subprocess.check_output(cmd, shell=True).decode().strip())
```
Comment (eranco74) [inline skipper/utils.py:147]:
```suggestion
# add this to utils:
def is_tool(name):
    """Check whether `name` is on PATH and marked as executable."""

    # from whichcraft import which
    from shutil import which

    return which(name) is not None
```
Than check if docker exists
if not check for podman
if both doesn't exists fail! 
https://stackoverflow.com/questions/11210104/check-if-a-program-exists-from-a-python-script

File: skipper/utils.py [hunk 4]
```diff
@@ -137,3 +137,20 @@ def image_to_dockerfile(image):
...
+        CONTAINER_RUNTIME_COMMAND = runtime
+    return CONTAINER_RUNTIME_COMMAND
+
+
+def run_container_command(args):
+    cmd = [get_runtime_command()]
+    cmd.extend(args)
+    print("BBBBBBBBBBB", cmd)
```
Comment (eranco74) [inline skipper/utils.py:155]: BBBBB?
```suggestion
    cmd = [get_runtime_command(), args]
```

File: skipper/runner.py [hunk 5]
```diff
@@ -62,8 +74,9 @@ def _run_nested(fqdn_image, environment, command, interactive, name, net, volume
     docker_cmd += ['-e', 'SKIPPER_UID=%(user_id)s' % dict(user_id=user_id)]
     docker_cmd += ['-e', 'HOME=%(homedir)s' % dict(homedir=homedir)]
 
-    docker_gid = grp.getgrnam('docker').gr_gid
-    docker_cmd += ['-e', 'SKIPPER_DOCKER_GID=%(docker_gid)s' % dict(docker_gid=docker_gid)]
+    if runtime == "docker":
+        docker_gid = grp.getgrnam('docker').gr_gid
+        docker_cmd += ['-e', 'SKIPPER_DOCKER_GID=%(docker_gid)s' % dict(docker_gid=docker_gid)]
```
[the remark this answers is not in the document]
Comment (tsorya) [reply skipper/runner.py:79]: it succeeded to run so i think it doesnt

File: skipper/utils.py (hunk 3 above)
Comment (tsorya) [reply skipper/utils.py:147]: much better, will change

File: skipper/runner.py [hunk 6]
```diff
@@ -19,8 +19,10 @@ def run(command, fqdn_image=None, environment=None, interactive=False, name=None
     return _run(command)
 
 
-def _run(cmd):
+def _run(cmd_args):
     logger = logging.getLogger('skipper')
+    cmd = [utils.get_runtime_command()]
+    cmd.extend(cmd_args)
```
Comment (eranco74) [inline skipper/runner.py:25]:
```suggestion
    cmd = [utils.get_runtime_command(), cmd_args]
```

**Pull request merged**